### PR TITLE
Help: with more help information (Issues #13)

### PR DESCRIPTION
**Pull request by Sn0rt**
This is a draft for #13 .
I intend to move to the code of help inside to the image package, Currently.
Signed-off-by: Sn0rt <wanggh-fnst@cn.fujitsu.com>

### CODE REVIEW

**Comment by wking** [comment]
On Tue, Nov 22, 2016 at 01:23:22AM -0800, Sn0rt wrote:
> This is a draft.  I intend to move to the code of help inside to the
> image package…

The draft looks good to me.  SpecURL and IssuesURL look like a
reasonable fit for image/{something}.go.  Maybe project.go?  Putting
a:

  func PrintHelp() {…}

in there would be good to, but I'd rather not import cobra in image.
The command wrappers (you'll want one for each command) would look
like:

  cmd.SetHelpFunc(func(c *cobra.Command, args []string) {
    image.PrintHelp()
  })

**Comment by Sn0rt** [comment]
@wking the image package is pure, so we should't import cobra packge in.
and this PR is updated.

File: cmd/oci-create-runtime-bundle/main.go [hunk 1]
```diff
@@ -91,6 +91,18 @@ It is strongly recommended to keep the default value.`,
...
+
+	cmd.SetHelpFunc(func(c *cobra.Command, args []string) {
+		err := cmd.Usage()
+		if err != nil {
+			stderr.Fatal(err)
+			os.Exit(0)
+		}
+		stdout.Println("\nMore information:")
+		stdout.Printf("\treferences %s\n", image.SpecURL)
+		stdout.Printf("\tBug report %s\n", image.IssuesURL)
```
Comment (wking) [inline cmd/oci-create-runtime-bundle/main.go]: I don't care which way the capitalization goes, but we should be consistent.

File: cmd/oci-create-runtime-bundle/main.go [hunk 2]
```diff
@@ -91,6 +91,18 @@ It is strongly recommended to keep the default value.`,
 		&v.version, "version", "v", false,
 		`Print version information and exit`,
 	)
+
+	cmd.SetHelpFunc(func(c *cobra.Command, args []string) {
+		err := cmd.Usage()
```
Comment (wking) [inline cmd/oci-create-runtime-bundle/main.go]: This doesn't show the long description like the standard help function does.  I think we want something like:

```go
origHelp := cmd.HelpFunc()
cmd.SetHelpFunc(func(c *cobra.Command, args []string) {
  origHelp(c, args)
  stdout.Println("\nMore information:")
  stdout.Printf("\tReferences %s\n", image.SpecURL)
  stdout.Printf("\tBug report %s\n", image.IssuesURL)
})
```

File: cmd/oci-create-runtime-bundle/main.go (hunk 1 above)
Comment (Sn0rt) [inline cmd/oci-create-runtime-bundle/main.go]: fixed

File: cmd/oci-create-runtime-bundle/main.go (hunk 2 above)
Comment (Sn0rt) [inline cmd/oci-create-runtime-bundle/main.go]: fixed

File: image/config.go [hunk 3]
```diff
@@ -31,6 +31,12 @@ import (
 	"github.com/pkg/errors"
 )
 
+// SpecURL is image-specs
+var SpecURL = "https://github.com/opencontainers/image-spec"
```
Comment (wking) [inline image/config.go]: `config.go` is about manipulating the image-config object, so I don't think these URLs are a good fit.  I expect we'll want a new file for them, and had [floated `project.go` earlier][1].

[1]: https://github.com/opencontainers/image-tools/pull/88#issuecomment-262326315

Comment (Sn0rt) [inline image/config.go]: `updated`

File: image/project.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package image
+
+// SpecURL is image-specs
```
Comment (wking) [inline image/project.go]: `image-specs` → `image-spec`, and probably worth having a full sentence for godoc:

    // SpecURL is the URL for the image-spec repository.

Comment (Sn0rt) [inline image/project.go]: `thx`

File: cmd/oci-create-runtime-bundle/main.go [hunk 5]
```diff
@@ -91,6 +91,16 @@ It is strongly recommended to keep the default value.`,
...
 		`Print version information and exit`,
 	)
+
+	origHelp := cmd.HelpFunc()
+
+	cmd.SetHelpFunc(func(c *cobra.Command, args []string) {
+		origHelp(c, args)
+		stdout.Println("\nMore information:")
+		stdout.Printf("\treferences\t%s\n", image.SpecURL)
+		stdout.Printf("\tbug reportt\t%s\n", image.IssuesURL)
```
Comment (Sn0rt) [inline cmd/oci-create-runtime-bundle/main.go]: `thx`

**Comment by wking** [comment]
9624729 looks good to me.

**Comment by coolljt0725** [comment]
LGTM

**Comment by xiekeyang** [comment]
LGTM

**Pull request merged**